PennMobile/Dining/Networking + Cache/DiningAPI.swift

```diff
@@ -137,3 +137,26 @@ extension DiningAPI {
         task.resume()
     }
 }
+
+// MARK: Current Dining Plan Start Date
+extension DiningAPI {
+    func getDiningPlanStartDate(diningToken: String, _ callback: @escaping (_ startDate: Date?) -> Void) {
```

Comment:
If possible, can you make this async/await instead of a callback? Here's an example of getPastDiningBalances using async/await:
```
func getPastDiningBalances(diningToken: String, startDate: String) async -> Result<[DiningBalance], NetworkingError> {
        ...
        guard let (data, response) = try? await URLSession.shared.data(for: request), let httpResponse = response as? HTTPURLResponse, httpResponse.statusCode == 200 else {
            return .failure(.serverError)
        }
        let decoder = JSONDecoder()
        decoder.keyDecodingStrategy = .convertFromSnakeCase
        if let decodedBalances = try? decoder.decode(PastDiningBalances.self, from: data) {
            return .success(decodedBalances.balanceList)
        } else {
            return .failure(.parsingError)
        }
    }
```
Message me if you need any help!

Reply:
Yea ill do that!